pennylane/templates/constructors/broadcast_double.py

```diff
@@ -0,0 +1,237 @@
+
+        .. code-block:: python
+
+            @template
+            def mytemplate(pars, wires):
+                qml.Hadamard(wires=wires[0])
+                qml.RY(pars[0], wires=wires[1])
+                qml.RX(pars[1], wires=wires[0])
+
+            @qml.qnode(dev)
+            def circuit(pars):
+                broadcast_double(block=mytemplate, wires=[0,1,2,3], parameters=pars)
+                return qml.expval(qml.PauliZ(0))
+
+            parameters_1 = [[1, 1]]
+            parameters_2 = [[2, 1]]
+            circuit([parameters_1, parameters_2])
+
+        If the number of parameters for a wire pair does not match the block's requirements, an error gets thrown:
+
+        .. code-block:: python
+
+                @template
+                def mytemplate(pars1, pars2, wires):
+                    qml.Hadamard(wires=wires)
+                    qml.RY(pars1, wires=wires)
+                    qml.RX(pars2, wires=wires)
+
+                @qml.qnode(dev)
+                def circuit(pars):
+                    broadcast_double(block=mytemplate, wires=[0, 1, 2, 3], parameters=pars)
+                    return qml.expval(qml.PauliZ(0))
+
+        >>> circuit([1, 2, 3])
+        TypeError: mytemplate() missing 1 required positional argument: 'pars2'
+
+        **Keyword arguments**
+
+        The block can be a template that takes additional keyword arguments.
+
+        .. code-block:: python
+
+            @template
+            def mytemplate(wires, h=True):
+                if h:
+                    qml.Hadamard(wires=wires[0])
+                qml.T(wires=wires[1])
+
+            @qml.qnode(dev)
+            def circuit(hadamard=None):
+                broadcast_double(block=mytemplate, wires=[0, 1, 2, 3], kwargs={'h': hadamard})
+                return qml.expval(qml.PauliZ(0))
+
+            circuit(hadamard=False)
+
+    """
+
+    #########
+    # Input checks
+
+    wires = _check_wires(wires)
+
+    _check_type(
+        parameters,
+        [Iterable, type(None)],
+        msg="'parameters' must be either of type None or "
+        "Iterable; got {}".format(type(parameters)),
+    )
+
+    if even:
+        n_pars = len(wires) // 2
+    else:
+        n_pars = (len(wires) - 1) // 2
+    if parameters is not None:
+        shape = _get_shape(parameters)
+        if shape[0] != n_pars:
+            raise ValueError(
+                "'parameters' must contain one entry for each of the {} wire pairs; got shape {}".format(
+                    n_pars, shape
+                )
+            )
+        # repackage for consistent unpacking
+        if len(shape) == 1:
+            parameters = [[p] for p in parameters]
+    else:
+        parameters = [[] for _ in range(len(wires))]
+
+    #########
+
+    if even:
+        start_with = 0
+    else:
+        start_with = 1
+
+    # extract the pairs of wires that the block acts on
+    wire_pairs = [[wires[i], wires[i + 1]] for i in range(start_with, len(wires) - 1, 2)]
```

Comment:
What happens if we have the same wire passed several times?

Reply:
Good point, never thought about this! I think this should be added to the wire test for all templates! Making a note.